hs_composite_resource/models.py

```diff
@@ -77,7 +77,10 @@ class name of that matching aggregation type.
             return None
 
         istorage = self.get_irods_storage()
-        store = istorage.listdir(dir_path)
+        irods_path = dir_path
+        if self.is_federated:
```

Comment:
Should federated path logic be handled in a different way?  I've seen a lot of these checks for is_federated and they're quite error prone... It'd be nice if we could encapsulate federation logic and avoid all of these federation checks.

We have FedStorage class which wraps IrodsStorage.  I think we could put some of this logic in there by implementing the IrodsStorage methods with federated path modifications.

Reply:
I thought root_path takes the federation into account?

Alternatively...
In FedStorage listdir could be overwritten to use the resource_federation_path (which could be passed to it in the constructor.

Reply:
@sblack-usu What I am trying to do -- at this point -- is to isolate these checks for federation to `BaseResource` and/or `AbstractResource`. Could not isolate them to `django_irods` because resource structure changes. In theory, this logic need not appear outside `BaseResource`.